src/cascade/model/covariates.py

```diff
@@ -0,0 +1,101 @@
+"""
+Represents covariates in the model.
+"""
+import logging
+from cascade.model.rates import Smooth
+
+
+CODELOG = logging.getLogger(__name__)
+MATHLOG = logging.getLogger(__name__)
+
+
+class CovariateColumn:
+    """
+    Establishes a reference value for a covariate column on input data
+    and in output data. It is possible to create a covariate column with
+    nothing but a name, but it must have a reference value before it
+    can be used in a model.
+
+    Args:
+        column_name (str): Name of hte column in the input data.
+        reference (float): Optional reference where covariate has no effect.
+        max_difference (float, optional):
+            If a data point's covariate is farther than `max_difference`
+            from the reference value, then this data point is excluded
+            from the calculation.
+    """
+    def __init__(self, column_name, reference=None, max_difference=None):
+        self._name = None
+        self._reference = None
+        self._max_difference = None
+
+        self.name = column_name
+        if reference is not None:
+            self.reference = reference
+        self.max_difference = max_difference
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, nom):
+        if not isinstance(nom, str):
+            raise ValueError(f"CovariateColumn name must be a string {nom}")
+        if len(nom) < 1:
+            raise ValueError(f"CovariateColumn name must not be empty string")
+        self._name = nom
+
+    @property
+    def reference(self):
+        return self._reference
+
+    @reference.setter
+    def reference(self, ref):
+        self._reference = float(ref)
+
+    @property
+    def max_difference(self):
+        return self._max_difference
+
+    @max_difference.setter
+    def max_difference(self, difference):
+        if difference is None:
+            self._max_difference = None
+        else:
+            diff = float(difference)
+            if diff <= 0:
```

Comment:
Brad says "This value must be greater than or equal zero"

Reply:
? What the heck does it mean if it's zero? OK, Brad.